packages/babel-plugin-transform-modules-commonjs/src/dynamic-import.ts

```diff
@@ -29,8 +28,10 @@ export function transformDynamicImport(
         Promise.resolve().then(() => ${buildRequire(source, file)})
       `
       : template.expression.ast`
-        new Promise(r => r(${source}))
-          .then(s => ${buildRequire(t.identifier("s"), file)})
+        (source =>
+          new Promise(r => r("" + source))
```

Comment:
```suggestion
          new Promise(r => r(${
            t.isTemplateLiteral(source) ? source : t.templateLiteral([], [source])
          }))
```

The transform from `${x}` to `"" + x` relies on the `ignoreToPrimitiveHint` assumption.

Consider an exotic object:

```js
var source = {
  [Symbol.toPrimitive](hint) {
    if (hint === 'string') {
      return "foo"
    }
    return null
  }
}
```

`"" + source` will be `null` while `${source}` will be `"foo"`. The [ToString](https://tc39.es/ecma262/#sec-tostring) operation should respect the `toPrimitive` symbol.

Reply:
Fixed, and I go further to move the common logic to function `buildDynamicImport`:

https://github.com/babel/babel/blob/75996fe68e4d00aed78e465f330fcb37a406d71f/packages/babel-helper-module-transforms/src/dynamic-import.ts#L18-L22

And in `transform-modules-amd/commonjs/system` we only need to write code once:

https://github.com/babel/babel/blob/75996fe68e4d00aed78e465f330fcb37a406d71f/packages/babel-plugin-transform-modules-amd/src/index.ts#L102-L114

https://github.com/babel/babel/blob/75996fe68e4d00aed78e465f330fcb37a406d71f/packages/babel-plugin-transform-modules-commonjs/src/dynamic-import.ts#L24-L26

https://github.com/babel/babel/blob/75996fe68e4d00aed78e465f330fcb37a406d71f/packages/babel-plugin-transform-modules-systemjs/src/index.ts#L277-L285